stream_alert/alert_processor/output_base.py

```diff
@@ -0,0 +1,319 @@
+            context = None
+            if not verify:
+                context = ssl.create_default_context()
+                context.check_hostname = False
+                context.verify_mode = ssl.CERT_NONE
+            request = urllib2.Request(url, data=data, headers=headers)
+            resp = urllib2.urlopen(request, context=context)
+            return resp
+        except urllib2.HTTPError as err:
+            raise OutputRequestFailure('Failed to send to {} - [{}] {}'.format(err.url,
+                                                                               err.code,
+                                                                               err.read()))
+    @staticmethod
+    def _check_http_response(resp):
+        return resp and (200 <= resp.getcode() <= 299)
+
+    @classmethod
+    def _get_default_properties(cls):
+        """Base method for retrieving properties that should be hard-coded for this
+        output service integration. This could include information such as a static
+        url used for sending the alerts to this service, a static port, or other
+        non-sensitive information.
+
+        If information of this sort is needed, this should be overridden in output subclasses.
+
+        NOTE: This should not contain any sensitive or use-case specific data. Information
+        such as this should be retrieved from the user using `get_user_defined_properties()`
+        so the user is prompted for the sensitive information at configuration time and said
+        information is then sent to kms for encryption and s3 for storage.
+
+        Returns:
+            [OrderedDict] Contains various OutputProperty items
+        """
+        pass
+
+    def get_secrets_bucket_name(self):
+        """Returns the streamalerts secrets s3 bucket name"""
+        return self._format_s3_bucket('streamalert.secrets')
+
+    def output_cred_name(self, descriptor):
+        """Formats the output name for this credential by combining the service
+        and the descriptor.
+
+        Args:
+            descriptor [string]: Service destination (ie: slack channel, pd integration)
+
+        Return:
+            [string] Formatted credential name (ie: slack_ryandchannel)
+        """
+        cred_name = str(self.__service__)
+
+        # should descriptor be enforced in all rules?
+        if descriptor:
+            cred_name = '{}_{}'.format(cred_name, descriptor)
+
+        return cred_name
+
+    def get_config_service(self):
+        """Get the string used for saving this service to the config. AWS services
+        are not named the same in the config as they are in the rules processor, so
+        having the ability to return a string like 'aws-s3' instead of 's3' is required
+
+        Returns:
+            [string] Service string used for looking up info in output configuration
+        """
+        return (self.__config_service__,
+                self.__service__)[self.__config_service__ == NotImplemented]
```

Comment:
Using a bool as an int? I actually didn't even know you could do that 😲 
This seems unnecessarily hacky. How about:

```python
return self.__service__ if self.__config_service__ == NotImplemented else self.__config_service__
```